config/c_get_alignment.m4

```diff
@@ -33,41 +34,29 @@ dnl
 # what we use. Yes, they should be the same - but "should" and
 # "are" frequently differ
 AC_DEFUN([OPAL_C_GET_ALIGNMENT],[
-    AC_CACHE_CHECK([alignment of $1],
-                   [AS_TR_SH([opal_cv_c_align_$1])],
-		   [AC_RUN_IFELSE([AC_LANG_PROGRAM([AC_INCLUDES_DEFAULT
-                                                    #include <stdbool.h> ],
-[[
-    struct foo { char c; $1 x; };
-    struct foo *p = (struct foo *) malloc(sizeof(struct foo));
-    int diff;
-    FILE *f=fopen("conftestval", "w");
-    if (!f) exit(1);
-    diff = ((char *)&p->x) - ((char *)&p->c);
-    free(p);
-    fprintf(f, "%d\n", (diff >= 0) ? diff : -diff);
-]])],                         [AS_TR_SH([opal_cv_c_align_$1])=`cat conftestval`],
-                               [AC_MSG_WARN([*** Problem running configure test!])
-                                AC_MSG_WARN([*** See config.log for details.])
-                                AC_MSG_ERROR([*** Cannot continue.])],
-                               [ # cross compile - do a non-executable test.  Trick
-                                 # taken from the Autoconf 2.59c.  Switch to using
-                                 # AC_CHECK_ALIGNOF when we can require Autoconf 2.60.
-                                 _AC_COMPUTE_INT([(long int) offsetof (opal__type_alignof_, y)],
-                                                 [AS_TR_SH([opal_cv_c_align_$1])],
-                                                 [AC_INCLUDES_DEFAULT
-#include <stdbool.h>
+    OPAL_VAR_SCOPE_PUSH([opal_align_name opal_align_value])
+    AC_LANG_PUSH([C])
 
-#ifndef offsetof
-# define offsetof(type, member) ((char *) &((type *) 0)->member - (char *) 0)
-#endif
-typedef struct { char x; $1 y; } opal__type_alignof_;
-],
-                                                 [AC_MSG_WARN([*** Problem running configure test!])
-                                                  AC_MSG_WARN([*** See config.log for details.])
-                                                  AC_MSG_ERROR([*** Cannot continue.])])])])
+    AC_CHECK_ALIGNOF([$1], [AC_INCLUDES_DEFAULT
+                            #include <stdbool.h>
+                           ])
 
-AC_DEFINE_UNQUOTED([$2], [$AS_TR_SH([opal_cv_c_align_$1])], [Alignment of type $1])
-eval "$2=$AS_TR_SH([opal_cv_c_align_$1])"
+    # Make a safe shell variable name (e.g., if $1 is "long long")
+    opal_align_name=AS_TR_SH([$1])
```

Comment:
You're no longer using opal_align_name, so lines 44-45 and the variable push can be removed.